otlp/common.go

```diff
@@ -580,3 +583,55 @@ func shouldTrimTraceId(traceID []byte) bool {
 	}
 	return true
 }
+
+func GetSampleRate(attrs map[string]interface{}) int32 {
+	sampleRateKey := GetSampleRateKey(attrs)
+	if sampleRateKey == "" {
+		return defaultSampleRate
+	}
+
+	sampleRate := defaultSampleRate
+	sampleRateVal := attrs[sampleRateKey]
+	switch v := sampleRateVal.(type) {
+	case string:
+		if i, err := strconv.Atoi(v); err == nil {
+			if i < math.MaxInt32 {
+				sampleRate = int32(i)
+			} else {
+				sampleRate = math.MaxInt32
+			}
+		}
+	case int32:
+		sampleRate = v
+	case int:
+		if v < math.MaxInt32 {
```

Comment:
if you're going to be this careful, you should probably guard against negative values too.